### PR TITLE
Make BecomeVisibleCustomAttribute use the IntersectionObserver API

### LINKED ISSUE #58: Use IntersectionObserver for BecomeVisibleCustomAttribute
The BecomeVisibleCustomAttribute uses [this implementation from StackOverflow](https://stackoverflow.com/a/125106).
In the question, [a more recent solution](https://stackoverflow.com/a/68262400) exists.
It employs the [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API).
I would advocate for the BecomeVisibleCustomAttribute to use this as well, as it simplifies the implementation and has led to better results in some recent Use Cases of mine.

If this isn't what is wanted, I'd at least update to the [getBoundingClientRect variant](https://stackoverflow.com/a/7557433).

### PR DESCRIPTION
**Pull request by conradstrassburger**
Fixes #58.

### CODE REVIEW

**Review by mreiche — APPROVED** [review]
We should have a test for that.

File: src/attributes/become-visible-custom-attribute.ts [hunk 1]
```diff
@@ -9,13 +9,14 @@ import {autoinject} from "aurelia-dependency-injection";
 export class BecomeVisibleCustomAttribute {
 
     private _visible:Boolean = false;
-    private readonly _checkVisibility:EventListener;
+    private readonly _checkVisibility: IntersectionObserverCallback
+    private observer: IntersectionObserver
```
Comment (Zsar) [inline src/attributes/become-visible-custom-attribute.ts:13]: May also be `readonly`.

File: src/attributes/become-visible-custom-attribute.ts [hunk 2]
```diff
@@ -9,13 +9,14 @@ import {autoinject} from "aurelia-dependency-injection";
 export class BecomeVisibleCustomAttribute {
 
     private _visible:Boolean = false;
-    private readonly _checkVisibility:EventListener;
+    private readonly _checkVisibility: IntersectionObserverCallback
```
Comment (Zsar) [inline src/attributes/become-visible-custom-attribute.ts:12]: Field can be downgraded to local variable in constructor body, can it not?

File: src/attributes/become-visible-custom-attribute.ts [hunk 3]
```diff
@@ -9,13 +9,14 @@ import {autoinject} from "aurelia-dependency-injection";
...
+    private observer: IntersectionObserver
 
     constructor(
         private readonly _element:Element
     ) {
-        this._checkVisibility = (ev:Event) => {
-            let isVisible = BecomeVisibleCustomAttribute.elementInViewport(this._element as HTMLElement);
+        this._checkVisibility = ([entry]) => {
+            let isVisible = entry.isIntersecting
```
Comment (Zsar) [inline src/attributes/become-visible-custom-attribute.ts:19]: Can be `const`.

File: src/attributes/become-visible-custom-attribute.ts [hunk 4]
```diff
@@ -25,38 +26,14 @@ export class BecomeVisibleCustomAttribute {
                 this._element.dispatchEvent(ev);
             }
         }
-    }
-
-    /**
-     * @see https://stackoverflow.com/questions/123999/how-to-tell-if-a-dom-element-is-visible-in-the-current-viewport
-     */
-    static elementInViewport(el) {
```
Comment (Zsar) [inline src/attributes/become-visible-custom-attribute.ts:33]: FWIW: I have not yet seen it to be incorrect, but it _has_ featured prominently in my profiling sessions.
Good riddance, I say!

**Review by Zsar — APPROVED** [review]

**Pull request merged**